### PR TITLE
Use less restrictve Dialer interface in DialWithDialer

### LINKED ISSUE #201: How would you use Socks5 dialer?
Is there a way implemented to connect via socks5?

### PR DESCRIPTION
**Pull request by foxcpp**
Closes #201.

### CODE REVIEW

**Comment by foxcpp** [comment]
Woops.

**Comment by foxcpp** [comment]
Okay, I fixed it.

File: client/client.go [hunk 1]
```diff
@@ -592,11 +592,16 @@ func Dial(addr string) (c *Client, err error) {
 	return
 }
 
+type Dialer interface {
+	// Dial connects to the given address via the proxy.
+	Dial(network, addr string) (c net.Conn, err error)
```
Comment (emersion) [inline client/client.go]: We can remove the name of the return values here, they don't provide any documentation benefits.

Comment (foxcpp) [reply client/client.go]: Done!

**Comment by emersion** [comment]
Thanks!

**Pull request merged**